src/scheduler/ExecGraph.cpp

```diff
@@ -25,6 +25,28 @@ int countExecGraphNodes(const ExecGraph& graph)
     return count;
 }
 
+std::set<std::string> getExecGraphHostsForNode(const ExecGraphNode& node)
+{
+    std::set<std::string> hostsForNode;
+    hostsForNode.insert(node.msg.executedhost());
+
+    if (!node.children.empty()) {
```

Comment:
I think this `if` is unnecessary (as `for(auto c : node.children)` will do nothing if it's empty).